From a5a099c74b4327a5774c8c1ee3a2220a18cdde51 Mon Sep 17 00:00:00 2001
From: ruslanbay <67730802+ruslanbay@users.noreply.github.com>
Date: Fri, 28 Nov 2025 13:18:57 +0100
Subject: [PATCH 08/16] fix: .bus.csi2.channel -> .bus.csi2.vc,
 .bus.csi2.data_type -> .bus.csi2.dt
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

drivers/media/i2c/crlmodule/crlmodule-core.c: In function ‘crlmodule_get_frame_desc’:
drivers/media/i2c/crlmodule/crlmodule-core.c:2366:32: warning: comparison between ‘enum v4l2_mbus_frame_desc_type’ and ‘enum crl_frame_desc_type’ [-Wenum-compare]
 2366 |                 if (desc->type == CRL_V4L2_MBUS_FRAME_DESC_TYPE_CSI2) {
      |                                ^~
drivers/media/i2c/crlmodule/crlmodule-core.c:2371:48: error: ‘struct v4l2_mbus_frame_desc_entry_csi2’ has no member named ‘channel’
 2371 |                         desc->entry[i].bus.csi2.channel = (u8)val;
      |                                                ^
drivers/media/i2c/crlmodule/crlmodule-core.c:2377:48: error: ‘struct v4l2_mbus_frame_desc_entry_csi2’ has no member named ‘data_type’
 2377 |                         desc->entry[i].bus.csi2.data_type = (u8)val;
---
 drivers/media/i2c/crlmodule/crlmodule-core.c |  4 ++--
 drivers/media/i2c/ti964.c                    |  2 +-
 drivers/media/pci/intel/ipu-isys-csi2.c      |  2 +-
 drivers/media/pci/intel/ipu-isys-video.c     |  2 +-
 include/media/v4l2-subdev.h                  | 18 +++++++++++++++++-
 5 files changed, 22 insertions(+), 6 deletions(-)

diff --git a/drivers/media/i2c/crlmodule/crlmodule-core.c b/drivers/media/i2c/crlmodule/crlmodule-core.c
index 9045f8889..06093c99a 100644
--- a/drivers/media/i2c/crlmodule/crlmodule-core.c
+++ b/drivers/media/i2c/crlmodule/crlmodule-core.c
@@ -2368,13 +2368,13 @@ static int crlmodule_get_frame_desc(struct v4l2_subdev *subdev,
 					sensor, crl_desc[i].csi2_channel, &val);
 			if (ret)
 				return ret;
-			desc->entry[i].bus.csi2.channel = (u8)val;
+			desc->entry[i].bus.csi2.vc = (u8)val;
 
 			ret = __crlmodule_parse_dynamic_entity(
 				      sensor, crl_desc[i].csi2_data_type, &val);
 			if (ret)
 				return ret;
-			desc->entry[i].bus.csi2.data_type = (u8)val;
+			desc->entry[i].bus.csi2.dt = (u8)val;
 		}
 	}
 
diff --git a/drivers/media/i2c/ti964.c b/drivers/media/i2c/ti964.c
index 7d73c80a7..06cec63b7 100644
--- a/drivers/media/i2c/ti964.c
+++ b/drivers/media/i2c/ti964.c
@@ -356,7 +356,7 @@ static int ti964_get_frame_desc(struct v4l2_subdev *sd,
 		entry->two_dim.width = ffmt->width;
 		entry->two_dim.height = ffmt->height;
 		entry->pixelcode = ffmt->code;
-		entry->bus.csi2.channel = vc++;
+		entry->bus.csi2.vc = vc++;
 		entry->bpp = csi_format->compressed;
 		entry++;
 	}
diff --git a/drivers/media/pci/intel/ipu-isys-csi2.c b/drivers/media/pci/intel/ipu-isys-csi2.c
index dd84d58ee..e08415a46 100644
--- a/drivers/media/pci/intel/ipu-isys-csi2.c
+++ b/drivers/media/pci/intel/ipu-isys-csi2.c
@@ -145,7 +145,7 @@ static int ipu_get_frame_desc_entry_by_dt(struct v4l2_subdev *sd,
 		return rval;
 
 	for (i = 0; i < desc.num_entries; i++) {
-		if (desc.entry[i].bus.csi2.data_type != data_type)
+		if (desc.entry[i].bus.csi2.dt != data_type)
 			continue;
 		*entry = desc.entry[i];
 		return 0;
diff --git a/drivers/media/pci/intel/ipu-isys-video.c b/drivers/media/pci/intel/ipu-isys-video.c
index e57b691b4..64723cda8 100644
--- a/drivers/media/pci/intel/ipu-isys-video.c
+++ b/drivers/media/pci/intel/ipu-isys-video.c
@@ -702,7 +702,7 @@ static int link_validate(struct media_link *link)
 		sd = media_entity_to_v4l2_subdev(ip->external->entity);
 		rval = ipu_isys_subdev_get_frame_desc(sd, &desc);
 		if (!rval && ip->stream_id < desc.num_entries)
-			ip->vc = desc.entry[ip->stream_id].bus.csi2.channel;
+			ip->vc = desc.entry[ip->stream_id].bus.csi2.vc;
 	}
 
 err_subdev:
diff --git a/include/media/v4l2-subdev.h b/include/media/v4l2-subdev.h
index 4bfeed294..98c1ff812 100644
--- a/include/media/v4l2-subdev.h
+++ b/include/media/v4l2-subdev.h
@@ -342,8 +342,13 @@ enum v4l2_mbus_frame_desc_flags {
  * struct v4l2_mbus_frame_desc_entry - media bus frame description structure
  *
  * @flags:	bitmask flags, as defined by &enum v4l2_mbus_frame_desc_flags.
+ * @bpp: bits per pixel
  * @pixelcode:	media bus pixel code, valid if @flags
  *		%FRAME_DESC_FL_BLOB is not set.
+ * @start_line: start line of the data for 2D DMA
+ * @start_pixel: start pixel of the data for 2D DMA
+ * @width: image width for 2D DMA
+ * @height: image height for 2D DMA
  * @length:	number of octets per frame, valid if @flags
  *		%V4L2_MBUS_FRAME_DESC_FL_LEN_MAX is set.
  * @bus:	Bus-specific frame descriptor parameters
@@ -351,8 +356,17 @@ enum v4l2_mbus_frame_desc_flags {
  */
 struct v4l2_mbus_frame_desc_entry {
 	enum v4l2_mbus_frame_desc_flags flags;
+	u8 bpp;
 	u32 pixelcode;
-	u32 length;
+	union {
+		struct {
+			u16 start_line;
+			u16 start_pixel;
+			u16 width;
+			u16 height;
+		} two_dim;
+		u32 length;
+	};
 	union {
 		struct v4l2_mbus_frame_desc_entry_csi2 csi2;
 	} bus;
@@ -378,7 +392,9 @@ struct v4l2_mbus_frame_desc_entry {
  */
 enum v4l2_mbus_frame_desc_type {
 	V4L2_MBUS_FRAME_DESC_TYPE_UNDEFINED = 0,
+	V4L2_MBUS_FRAME_DESC_TYPE_PLATFORM,
 	V4L2_MBUS_FRAME_DESC_TYPE_PARALLEL,
+	V4L2_MBUS_FRAME_DESC_TYPE_CCP2,
 	V4L2_MBUS_FRAME_DESC_TYPE_CSI2,
 };
 
-- 
2.34.1

